### PR TITLE
chore: Fix docs for latest version of typedoc

### PR DESCRIPTION
**Pull request by marikaner**
Use the better support of es6 modules with typedoc 0.20. Remove lerna plugin and fix links.

### CODE REVIEW

File: packages/core/src/odata-common/filter/filterable.ts [hunk 1]
```diff
@@ -27,7 +27,7 @@ export type Filterable<EntityT extends Entity> =
  *  .filter(and(filterExp1, filterExp2));
  * ```
  *
- * Note that the [[GetAllRequestBuilderV2.filter]]  and [[GetAllRequestBuilderV4.filter]] method take a rest parameter and thereby an array of filter expressions that are then combined conjunctively. As a consequence following is equivalent to the example above:
+ * Note that the [[GetAllRequestBuilder.filter | GetAllRequestBuilderV2.filter]]  and [[GetAllRequestBuilderV4.filter]] method take a rest parameter and thereby an array of filter expressions that are then combined conjunctively. As a consequence following is equivalent to the example above:
```
Comment (jjtang1985) [inline packages/core/src/odata-common/filter/filterable.ts:30]: `[[GetAllRequestBuilder.filter | GetAllRequestBuilderV2.filter]]` is the key to fix the reference?

File: package.json [hunk 2]
```diff
@@ -1,5 +1,6 @@
 {
   "name": "sap-cloud-sdk",
+  "version": "1.34.0",
```
Comment (jjtang1985) [inline package.json:3]: Is this mandatory for the typedoc, since lerna is not used?

**Review by jjtang1985 — APPROVED** [review]
Looks good. I adjusted the script. Please have a check.

File: package.json (hunk 2 above)
Comment (marikaner) [reply package.json:3]: Yes.

File: packages/core/src/odata-common/filter/filterable.ts (hunk 1 above)
Comment (marikaner) [reply packages/core/src/odata-common/filter/filterable.ts:30]: Yes.

File: scripts/generate-docs.js [hunk 3]
```diff
@@ -50,7 +50,7 @@ function adjustForGitHubPages() {
 }
 
 function adjustSearchJs(paths) {
-  const filtered = paths.filter(isSearchJson);
+  const filtered = paths.filter(isSearchJs);
```
Comment (marikaner) [inline scripts/generate-docs.js:53]: BTW, we might want to not filter it here, right?

**Pull request merged**